code/Report.php

```diff
@@ -349,7 +349,7 @@ public function getReportField()
         if (Injector::inst()->has(HTTPRequest::class)) {
             /** @var HTTPRequest $request */
             $request = Injector::inst()->get(HTTPRequest::class);
-            $params = $request->param('filters') ?: [];
+            $params = $request->param('filters') ?: $request->getVar('filters') ?: [];
```

Comment:
Is this safe? Does `sourceRecords` expect to receive raw request vars?

Reply:
Well it definitely used to work in elemental on SS4. Perhaps I should try and find where the regression happened before we go forward with this

Reply:
Yes please

Reply:
The two (non-elemental) instances of using `filters` via $params in CWP 2.1 are here:

* https://github.com/silverstripe/silverstripe-cms/blob/4.2/code/Reports/BrokenLinksReport.php#L105-L109
* https://github.com/silverstripe/silverstripe-sitewidecontent-report/blob/3.0/src/SitewideContentReport.php#L205

Both of these use the `$_REQUEST` superglobal, which is probably because $params wasn't correctly passed to the methods (relevant fix was merged in #106).

I guess I don't have a good comparison to use for where $params would come from URL parameters. In light of this, perhaps using getVars is OK?

I'm not sure how this ever worked in elemental to be honest.

Reply:
Using the broken links report as an example, `filter` is a get var AND a post var when you make the form submission, but not sure that it would get into the params

Reply:
I think this was clearly the original intention. Tracing back that LOC, it was originally using the `$_REQUEST` superglobal until late last year: https://github.com/silverstripe/silverstripe-reports/pull/88/files

Reply:
Hiya team, 

I can confirm that this fix works for me. 

I'm currently playing around with the filters in the `Pages due for review` report which involves some date filtering and filtering based off pages assigned to the currently logged in user.

Prior to this fix I can see that when the filter is triggered, it sends the correct set of query parameters however it appears that it is not extracting the data from the request correctly so this fix makes sense to me :) 

Please merge 👍 👍 👍